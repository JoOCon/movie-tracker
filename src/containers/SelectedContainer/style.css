.selected-container {
  margin: 1.5em 0;
  display: flex;
  border: 2px solid #ffcb9a;
  background-color: #116466;
  justify-content: space-between;
  width: 60%;
  font-family: "Bangers", cursive;
  transition: 600ms; }

.selected-container h1 {
  color: #ffcb9a;
  font-size: 3em;
  text-shadow: 2.5px 2.5px black;
  margin: 0; }

.selected-container p {
  margin: 0;
  font-size: 1.2em; }

.selected-movie-content {
  margin: auto;
  padding: 0 1em; }

.selected-wrapper {
  display: flex;
  justify-content: center; }
