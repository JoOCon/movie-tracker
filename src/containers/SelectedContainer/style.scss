@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';

.selected-container {
  margin: 1.5em 0;
  display: flex;
  border: 2px solid $gold;
  background-color: $turquoise;
  justify-content: space-between;
  width: 60%;
  font-family: $title-font;
  transition: 600ms;
}

.selected-container h1 {
  color: $gold;
  font-size: 3em;
  text-shadow: 2.5px 2.5px black;
  margin: 0;
}

.selected-container p {
  margin: 0;
  font-size: 1.2em;
}

.selected-movie-content {
  margin: auto;
  padding: 0 1em;
}

.selected-wrapper {
    display: flex;
    justify-content: center;
}
