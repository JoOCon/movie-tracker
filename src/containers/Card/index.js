import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import { addFavorite } from '../../actions';
import { postFavorites } from '../../api/apiCalls';
import './style.css';

export const Card = (props) => {
  const { movie, user, addFavorite } = props;
  const { poster_path, vote_average } = movie;
  const rating = vote_average / 2;

  const handleFavorite = () => {
    const favoriteData = {...movie, user_id: user.id};
    addFavorite(favoriteData);
    postFavorites(favoriteData);
  };

  return (
    <div className='card'>
      <div className='favorite-wrapper'>
        <button 
          className='favorite' 
          onClick={() => handleFavorite()}>
        </button>
        <StarRatings 
          starDimension={'1em'} 
          rating={rating} 
          numberOfStars={5} 
          starRatedColor={'gold'} 
        />
      </div>
      <img className='poster-image' src={poster_path} alt='movie data' />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userData
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  postFavorites: (movie) => dispatch(postFavorites(movie))
});

Card.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object,
  addFavorite: PropTypes.func,
  postFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
