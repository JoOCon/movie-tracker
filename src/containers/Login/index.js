import React, { Component } from 'react';
import './style.css';
import { login } from '../../api/apiCalls'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await login(this.state);
    // this.setState({ userName: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='email' placeholder='email' type='email' value={email} onChange={this.handleChange} />
        <input name='password' placeholder='password' type='password' value={password} onChange={this.handleChange} />
        <button>LogIn</button>
      </form>
    );
  }
}
