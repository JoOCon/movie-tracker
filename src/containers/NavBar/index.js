import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};