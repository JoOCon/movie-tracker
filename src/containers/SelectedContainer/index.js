import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { manageSelectedMovie } from '../../actions';
import './style.css';

export const SelectedContainer = (props) => {
  const { selectedMovie, manageSelectedMovie } = props;
  const {
    overview,
    poster_path,
    release_date,
    title
  } = props.selectedMovie;

  const displayInfo = () => { 
    if (poster_path){
      return (
        <div className='selected-container' onClick={() => manageSelectedMovie(selectedMovie)}>
          <img className='poster-image' src={poster_path} alt='movie data' />
          <div className='selected-movie-content'>
            <h1>{title}</h1>
            <h3>RELEASE: {release_date}</h3>
            <p>DETAILS: {overview}</p>
          </div>
        </div>);
    } else {
      return <div></div>;
    }
  };

  return (
    <div className='selected-wrapper'>{displayInfo()}</div>
  );
};

export const mapStateToProps = (state) => ({
  selectedMovie: state.selectedMovie
});

export const mapDispatchToProps = (dispatch) => ({
  manageSelectedMovie: (movie) => dispatch(manageSelectedMovie(movie))
});

SelectedContainer.propTypes = {
  selectedMovie: PropTypes.object,
  manageSelectedMovie: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(SelectedContainer);
