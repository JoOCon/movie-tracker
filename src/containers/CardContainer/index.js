import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import './style.css';

const CardContainer = (props) => {
  const displayMovies = props.movies.map(movie => <Card key={movie.id} movie={movie} />);
  return (
    <div className="card-container">
      {displayMovies}
    </div>
  );
};

const mapStateToProps = (state) => ({
  movies: state.moviesData
});

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps)(CardContainer);
