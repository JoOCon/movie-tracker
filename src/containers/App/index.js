import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { addMovies } from '../../actions';
import { fetchMovies } from '../../api/apiCalls';
import CardContainer from '../CardContainer';
import FavoriteContainer from '../FavoriteContainer';
import Login from '../../containers/Login';
import SignUp from '../../containers/SignUp';
import { Header } from '../../components/Header';
import './style.css';

export class App extends Component {
  componentDidMount = () => {
    this.fetchOpeningMovies();
  }

  fetchOpeningMovies = async () => {
    const { addMovies } = this.props;
    const openingMovies = await fetchMovies();
    addMovies(openingMovies);
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Route path='/' component={Header} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/signup' component={SignUp} />
          <Route exact path='/' component={CardContainer} />
          <Route exact path='/:landingPage' component={CardContainer} />
          <Route exact path='/:user/favorites' component={FavoriteContainer} />
        </div>
      </BrowserRouter>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (moviesData) => dispatch(addMovies(moviesData))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default connect(null, mapDispatchToProps)(App);
