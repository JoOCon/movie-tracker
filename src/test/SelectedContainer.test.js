import React from 'react';
import { mapStateToProps, SelectedContainer } from '../containers/SelectedContainer';
import { mockSelectedMovie } from '../test/mockData'
import { shallow } from 'enzyme';

describe('FavoriteContainer tests', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SelectedContainer selectedMovie={mockSelectedMovie}
      />
    );
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should have a selectedMovie in props', () => {
      const mockState = { selectedMovie: mockSelectedMovie };
      const expected = { selectedMovie: mockSelectedMovie };
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });
});

