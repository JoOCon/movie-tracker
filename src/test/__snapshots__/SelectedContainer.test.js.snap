// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectedMovie={
        Object {
            "movie_id": 353081,
            "overview": "When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.",
            "poster_path": "https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
            "release_date": "2018-07-25",
            "title": "Mission: Impossible - Fallout",
            "vote_average": 7.3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="selected-container"
        onClick={[Function]}
>
        <img
                alt="movie data"
                className="poster-image"
                src="https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
        />
        <div
                className="selected-movie-content"
        >
                <h1>
                        Mission: Impossible - Fallout
                </h1>
                <h3>
                        RELEASE: 
                        2018-07-25
                </h3>
                <p>
                        DETAILS: 
                        When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
                </p>
        </div>
</div>,
      "className": "selected-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie data"
            className="poster-image"
            src="https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
/>,
          <div
            className="selected-movie-content"
>
            <h1>
                        Mission: Impossible - Fallout
            </h1>
            <h3>
                        RELEASE: 
                        2018-07-25
            </h3>
            <p>
                        DETAILS: 
                        When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
            </p>
</div>,
        ],
        "className": "selected-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie data",
            "className": "poster-image",
            "src": "https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Mission: Impossible - Fallout
</h1>,
              <h3>
                RELEASE: 
                2018-07-25
</h3>,
              <p>
                DETAILS: 
                When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
</p>,
            ],
            "className": "selected-movie-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mission: Impossible - Fallout",
              },
              "ref": null,
              "rendered": "Mission: Impossible - Fallout",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "RELEASE: ",
                  "2018-07-25",
                ],
              },
              "ref": null,
              "rendered": Array [
                "RELEASE: ",
                "2018-07-25",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "DETAILS: ",
                  "When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "DETAILS: ",
                "When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="selected-container"
          onClick={[Function]}
>
          <img
                    alt="movie data"
                    className="poster-image"
                    src="https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
          />
          <div
                    className="selected-movie-content"
          >
                    <h1>
                              Mission: Impossible - Fallout
                    </h1>
                    <h3>
                              RELEASE: 
                              2018-07-25
                    </h3>
                    <p>
                              DETAILS: 
                              When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
                    </p>
          </div>
</div>,
        "className": "selected-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="movie data"
              className="poster-image"
              src="https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
/>,
            <div
              className="selected-movie-content"
>
              <h1>
                            Mission: Impossible - Fallout
              </h1>
              <h3>
                            RELEASE: 
                            2018-07-25
              </h3>
              <p>
                            DETAILS: 
                            When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
              </p>
</div>,
          ],
          "className": "selected-container",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie data",
              "className": "poster-image",
              "src": "https://image.tmdb.org/t/p/w200/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Mission: Impossible - Fallout
</h1>,
                <h3>
                  RELEASE: 
                  2018-07-25
</h3>,
                <p>
                  DETAILS: 
                  When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.
</p>,
              ],
              "className": "selected-movie-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mission: Impossible - Fallout",
                },
                "ref": null,
                "rendered": "Mission: Impossible - Fallout",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "RELEASE: ",
                    "2018-07-25",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "RELEASE: ",
                  "2018-07-25",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "DETAILS: ",
                    "When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "DETAILS: ",
                  "When an IMF mission ends badly, Thunted by assassins while trying to prevent a global catastrophe.",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
